jgrapht-core/src/main/java/org/jgrapht/alg/interval/LexBreadthFirstSearch.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * (C) Copyright 2018-2018, by Oliver Feith, Dennis Fischer and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.interval;
+
+import org.jgrapht.*;
+import org.jgrapht.traverse.*;
+import org.jgrapht.util.Ordering;
+
+import java.util.*;
+
+/** A class that is used to perform the BFS algorithms used to detect interval graphs.
+ * @author Oliver Feith
+ * @author Dennis Fischer
+ */
+public final class LexBreadthFirstSearch
```

Comment:
The class definitely shouldn't be public as it contains only auxiliary functions and can't be used by users.